dash/testing/browser.py

```diff
@@ -108,8 +112,26 @@ def percy_snapshot(self, name="", wait_for_callbacks=False):
             name, sys.version_info.major, sys.version_info.minor
         )
         logger.info("taking snapshot name => %s", snapshot_name)
-        if wait_for_callbacks:
-            until(self._wait_for_callbacks, timeout=10)
+        try:
+            if wait_for_callbacks:
+                until(
+                    self._wait_for_callbacks,
+                    timeout=40,
+                    poll=0.5,
+                    sleep_first=True,
```

Comment:
I'm not so sure about adding `sleep_first`... it means you always get `poll` amount of sleep time, whereas you could get the same effect without having to set an otherwise-unnecessarily large `poll` if you just put `time.sleep(0.5)` before the `until` call, right? That feels clearer to read as well.

Reply:
ah, I forgot to add the flag in the `percy_snapshot` API. my initial intention is to make it more flexible, so for some `visit_and_snapshot` calls, we should let it do a `sleep` before the first `check` to have some safe margins with `wait_for_callbacks`. I will revise it

Reply:
OK - if you do want to keep `sleep_first` as part of `until`, perhaps better would be to make it a number instead of a boolean, so it would be decoupled from `poll`.

Reply:
yes,  revisit the code, inclines to add hard sleep than sleep switch, and I will just use `wait_for_callbacks` to control that.